pkg/spdk/replica.go

```diff
@@ -1964,8 +1965,22 @@ func (r *Replica) rebuildingDstShallowCopyPrepare(spdkClient *spdkclient.Client,
 		return "", fmt.Errorf("cannot find snapshot %s in the rebuilding snapshot list for replica %s shallow copy prepare", snapshotName, r.Name)
 	}
 	if srcSnapSvcLvol.Parent != "" {
-		if r.BackingImage != nil && r.BackingImage.Name == srcSnapSvcLvol.Parent {
-			dstSnapshotParentLvolName = srcSnapSvcLvol.Parent
+		if IsBackingImageSnapLvolName(srcSnapSvcLvol.Parent) {
+			if r.BackingImage == nil {
+				return "", fmt.Errorf("found a src snapshot lvol %s has parent backing image %s but the dst backing image lvol is nil for replica %s shallow copy prepare", srcSnapSvcLvol.Name, srcSnapSvcLvol.Parent, r.Name)
```

Comment:
Maybe we can add `rebuilding snapshot %s` to the message as row 1978

Reply:
Sure. Addressed. Thank you!